### PR TITLE
Decode into a array instead of a slice

### PR DESCRIPTION
**Pull request by akalin-keybase**
This makes sure we try to decode only the first element of the args we receive.

### CODE REVIEW

**Comment by akalin-keybase** [comment]
cc @jzila also

File: test/files/sample_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+	protocol := SampleProtocol(nil)
+	processBigBytesDesc := protocol.Methods["processBigBytes"]
+	arg2 := processBigBytesDesc.MakeArg()
+
+	decoder := codec.NewDecoderBytes(b, &h)
+
+	// Test shouldn't hang here (i.e., it shouldn't try to decode
+	// all but the first argument element of badArgs).
+	err = decoder.Decode(arg2)
```
Comment (akalin-keybase) [inline test/files/sample_test.go]: Ideally our RPC error would reject requests with 'extra' args, but that can be done later.

**Comment by jzila** [comment]
But decoding is based on length, so we'd do that no matter what...

**Comment by akalin-keybase** [comment]
ugorji/codec decodes in place, so if the target type is a slice, it'll expand it. If it's an array, it'll decode only up to the array length (and skip the rest, with an option to error out that we don't use).

**Comment by jzila** [comment]
@akalin-keybase neat, thanks!

**Pull request merged**